### PR TITLE
Feature/area bar chart state handling

### PR DESCRIPTION
**Pull request by ochococo**
- Improved visually the **Loading** state.
- Added **Failed** state support.

simulator screen shot 17 04 2016 11 52 07
simulator screen shot 17 04 2016 11 52 19

### CODE REVIEW

File: GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift [hunk 1]
```diff
@@ -100,15 +103,27 @@ final class AreaBarChartView : UIView, ViewModelRendering {
...
         switch (state, toState) {
+        case (.Waiting, .Rendering(let viewModel)):
+            configureWithViewModel(viewModel)
+
+            UIView.animateWithDuration(1) {
+                self.informationSheet.alpha = 0
+            }
+
         case (_, .Rendering(let viewModel)):
             configureWithViewModel(viewModel)
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift]: What if we transition from .Failed to .Rendering ? Is it possible (information sheet would still be visible I think)?

**Comment by michallaskowski** [comment]
@ochococo That's all :)

File: GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift (hunk 1 above)
Comment (ochococo) [inline GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift]: Excellent point! Thanks! 👏 :

**Comment by michallaskowski** [comment]
Looks good!

**Pull request merged**